lib/octokit/client.rb

```diff
@@ -2,7 +2,6 @@
 require 'octokit/warnable'
 require 'octokit/arguments'
 require 'octokit/repo_arguments'
-require 'octokit/configurable'
 require 'octokit/authentication'
 require 'octokit/gist'
 require 'octokit/rate_limit'
```

Comment:
Should we move these to autoloads in the `Octokit` module as well in case they are referenced before `Client`?

Reply:
Right... it depends on the API of the gem. It looks like they are all public API and you expect these modules to be pluggable by other client types. 

I'll take a look at the modules that aren't under `Client` and see if it makes sense to autoload them. If they are trivial enough, we could consider just moving the require.

Reply:
I decided to autoload all constants nested in `Octokit` besides `Configurable` (because that's used immediately in the main module). It should be roughly the same performance diff that way.